### ISSUE TITLE
Can't find tag retention based on labels, despite screenshots showing this feature

### ISSUE THREAD

**Issue by MichaelKueller** [goharbor/harbor#11822]
We want to setup tag retention policies based on labels, which we set as part of our deployment pipleline. That way we can identify which images have been deployed to production and make sure not to delete the last n versions that have been deployed to production. This is in contrast to keeping the last n images pushed to harbor, as none of the last n images might have been pushed to production. 

In the screenshot below (found in https://github.com/goharbor/harbor/issues/8695) I saw that such a feature exists, but in my harbor installation (v. 1.9.4) the field to define labels is missing. Has this feature been removed, or where can I find it? 
![image](https://user-images.githubusercontent.com/784735/80943580-ce84e880-8de7-11ea-9e98-fcd8e4cbb700.png)

**Comment by xaleeks** [comment]
@MichaelKueller  Labels as a filter condition was removed shortly after it was released, it was causing some issues because of how they are created at system and project level but then interacted with the repository matching conditions, giving us some nasty side effects.  We will eventually add labels back when we have bandwidth for testing.  I've added this as a requirement to the backlog.

I'm curious if labels are enough for tracking image deployed in the wild like you described. ie how do you enforce that they are added when they are deployed and removed when they are taken down.

**Comment by MichaelKueller** [comment]
Thank you for your response. 
In our case our deployment tool adds the labels during the deployment process. In harbor we would then set up a rule to retain the newest 5 images that are labelled with "deployed-to-production". So there is no need to remove the labels. 
If labels are not available for retention policies, how would you suggest to keep the last 5 images that have been deployed to production, additionally to the newest 5 images?

**Comment by claflico** [comment]
i would like to be able to delete by labels as well please.

**Comment by Andreiaotto** [comment]
We also have this use case, we want to migrate from quay.io to harbor, as we have commitID as tag, we don't know what's the tag name to add rules.
With labels we can define the label in our pipelines and define retention rules.

Is there any recommendation while there is no such feature available?

thank you!

**Comment by kitenco** [comment]
we need labels for retention rules too, same use case (deployment to production)

**Comment by fabricev** [comment]
Agreed with everbody above, retentions policies based on labels would be really nice.

**Comment by anatoly-kryzhanovsky** [comment]
+1

**Comment by romanglogowski-millennium** [comment]
+1

**Comment by MichaelKueller** [comment]
Hi @xaleeks 
are you able to give an update about this feature? Is it going to return in one of the next releases? The lack of it is starting to cause quite some extra manual work and looking at the comments above it seems we are not the only ones who need it.

**Comment by kitenco** [comment]
Hi @xaleeks
are you able to give an update about this feature?

**Comment by Yingrjimsch** [comment]
Is this planned? We would like to use that too?

**Comment by Kajot-dev** [comment]
This feature would be nice. For example we would have a service which scans kubernetes/OKD cluster for images which are in use and labels them so, they won't get deleted by retention policies and this way the service would be independent from tags (and reduced risk from overlapping the tag names with users/CI/CD

**Comment by akramadel** [comment]
Is there a plan to release this feature soon?

**Comment by spirit1431007** [comment]
Any plan about this feature?

**Comment by marcelodefensepoint** [comment]
Any plan about this feature?

**Comment by crossRT** [comment]
configure retention condition by label is very useful.
because we can use it to identify production images and prevent it from being deleted.
is there any plan to bring this feature back?

**Comment by jon-nfc** [comment]
Are there any update to when this is going to be re-implemented?
